src/dataflow/components/dataflow-program.tsx

```diff
@@ -180,6 +180,7 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
   private deleteSelectedNodes = () => {
     this.programEditor.selected.list.forEach((n: Node) => {
       this.programEditor.removeNode(n);
+      this.programEditor.selected.clear();
```

Comment:
Probably doesn't matter, but conceptually it might be clearer to clear the selection before removing the node.

Reply:
Sounds good!

Reply:
Although if we end up handling multiple nodes selected at the same time, maybe the selection clear should be outside the forEach.

Reply:
I'm going to look at it a little closer and try to see what happens in the case where we have multiple nodes selected (not possible with current UI, but it is a request in PT).

Reply:
Also, not sure if any advantage between:
`this.programEditor.selected.list = [];`
and...
`this.programEditor.selected.clear();`
Both work fine.

I'll look at it again with a clear head in the morning.

Reply:
I'd vote for `clear()`ing the selection before the `forEach` loop.

Reply:
I'd vote for `clear()`ing the selection before the `forEach` loop.

Reply:
Has to be after of the forEach takes no action.

Reply:
Also was able to test with multi-select.  Did some research and ctrl + click allows for multi-node selection.

Reply:
Now that I look at it more closely, it makes sense that it has to be after. But presumably it should still be outside the loop.

Reply:
The theoretical concern is that it seems odd for `programEditor.selected` to contain references to nodes that have been removed, as would be true in the middle of the loop. One might imagine that in the future `programEditor.removeNode()` might remove any references to that node, such as those in `programEditor.selected`. One way to deal with that would be to copy the contents of `programEditor.selected` and iterate through the copy. Or maybe we just don't worry about it for now. 😉

Reply:
I agree. I find it odd that the Rete editor itself leaves a node in the selected list after it's been removed.  Those are Rete calls though so it's something happening on there end.  But, yes, ideally a call to `removeNode` should take the darn node out of the selected list (which is why I didn't catch the bug at first).

Reply:
This would be another case for a potential Rete PR.

Reply:
Given that it could be considered a bug, if they fix the bug it could potentially affect our loop, which might be an argument for iterating through a copy -- as well as possibly submitting a PR.

Reply:
Agree.  I think I will make a copy and iterate over that in case this changes in the future.